File: elixir/sublist/sublist.exs
defmodule Sublist do
  @doc """
  Returns whether the first list is a sublist or a superlist of the second list
  and if not whether it is equal or unequal to the second list.
  """
  def compare(a, a),      do: :equal
  def compare([], [nil]), do: :sublist
  def compare([nil], []), do: :superlist

  def compare(a, b) do
    s_a = "," <> Enum.join(a, ",") <> ","
    s_b = "," <> Enum.join(b, ",") <> ","
    cond do
      String.contains?(s_b, s_a) -> :sublist
      String.contains?(s_a, s_b) -> :superlist
      true                       -> :unequal
    end
  end
end

File: python/sublist/sublist.py
EQUAL     = 0
SUBLIST   = 1
SUPERLIST = 2
UNEQUAL   = 3

def check_lists(l1, l2):
    if l1 == l2:
        return EQUAL
    if sublist(l1, l2):
        return SUBLIST
    if sublist(l2, l1):
        return SUPERLIST
    return UNEQUAL


def sublist(l1, l2):
    return any(l1 == l2[i:i+len(l1)]
            for i in range(0, len(l2) - len(l1) + 1))
